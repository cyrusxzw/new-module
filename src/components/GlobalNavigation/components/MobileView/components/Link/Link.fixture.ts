import type { LinkProps } from './Link.types';

const LinkFixture: LinkProps = {
  id: 'link',
  label: 'Link',
  title: 'Link',
  url: '#link',
};

export { LinkFixture };
