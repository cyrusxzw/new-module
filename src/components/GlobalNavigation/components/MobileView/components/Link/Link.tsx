import React from 'react';
import cx from 'classnames';
import { useThemeContext } from '~/contexts';
import { Hyperlink } from '~/components/Hyperlink';
import { Icon } from '~/components/Icon';
import type { LinkType } from './Link.types';
import compositionStyles from '../../MobileView.module.css';
import styles from './Link.module.css';

const Link: LinkType = ({
  isTop,
  isNested,
  isVisible,
  label,
  title,
  url,
  isExternal,
}) => {
  const currentTheme = useThemeContext(null, 'dark');

  const classSet = cx(
    styles.base,
    styles[currentTheme],
    compositionStyles.itemElement,
    { [styles.top]: isTop },
    { [styles.nested]: isNested },
  );

  return (
    <Hyperlink
      className={classSet}
      tabIndex={!isVisible ? -1 : null}
      title={title}
      url={url}
    >
      {label}
      {isExternal && (
        <>
          {' '}
          <Icon height={14} name="rightArrow" theme={currentTheme} width={14} />
        </>
      )}
    </Hyperlink>
  );
};

export { Link };
