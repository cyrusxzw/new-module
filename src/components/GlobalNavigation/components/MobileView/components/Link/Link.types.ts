import type { ComponentWithoutChildren } from '~/types';
import type { Clickable } from '~/components/GlobalNavigation/GlobalNavigation.types';

type LinkProps = Clickable & {
  isNested?: boolean;
  isTop?: boolean;
  isVisible?: boolean;
  url: string;
};

type LinkType = ComponentWithoutChildren<LinkProps>;

export type { LinkType, LinkProps };
