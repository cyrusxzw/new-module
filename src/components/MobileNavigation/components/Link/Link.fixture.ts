const LinkFixture = {
  id: 'link',
  label: 'Link',
  title: 'Link',
  type: 'link' as const,
  url: '#link',
};

export { LinkFixture };
