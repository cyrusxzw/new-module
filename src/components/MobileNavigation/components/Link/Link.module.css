@import 'styles/index';

.base {
  &:not(.top) {
    padding-left: 42px;
  }
}

.nested.nested {
  padding-left: 72px;
}
