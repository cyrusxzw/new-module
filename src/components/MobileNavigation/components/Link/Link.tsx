import React from 'react';
import cx from 'classnames';
import { useThemeContext } from '~/contexts';
import { Hyperlink } from '~/components/Hyperlink';
import type { LinkType } from './Link.types';
import compositionStyles from '../../MobileNavigation.module.css';
import styles from './Link.module.css';

const Link: LinkType = ({ isTop, isNested, isVisible, label, title, url }) => {
  const currentTheme = useThemeContext(null, 'dark');
  const classSet = cx(
    styles.base,
    styles[currentTheme],
    compositionStyles.itemElement,
    { [styles.top]: isTop },
    { [styles.nested]: isNested },
  );

  return (
    <Hyperlink
      className={classSet}
      tabIndex={!isVisible ? -1 : null}
      title={title}
      url={url}
    >
      {label}
    </Hyperlink>
  );
};

export { Link };
