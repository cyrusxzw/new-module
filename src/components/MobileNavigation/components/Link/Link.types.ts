import type { ComponentWithoutChildren } from '~/types';

type LinkProps = {
  isNested?: boolean;
  isTop?: boolean;
  isVisible?: boolean;
  label: string;
  title: string;
  url: string;
};

type LinkType = ComponentWithoutChildren<LinkProps>;

export type { LinkType };
