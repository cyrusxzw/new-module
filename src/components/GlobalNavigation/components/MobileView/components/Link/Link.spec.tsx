import React from 'react';
import { render, screen } from '@testing-library/react';
import { Link } from './Link';
import { LinkFixture } from './Link.fixture';

describe('<GlobalNavigation.MobileView.Link />', () => {
  it.todo('should assert something');

  it('should render base component correctly', () => {
    const { container } = render(<Link {...LinkFixture} />);

    expect(container).toMatchSnapshot();
  });
});
