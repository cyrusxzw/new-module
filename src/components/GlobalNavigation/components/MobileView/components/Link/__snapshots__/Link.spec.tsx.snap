// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GlobalNavigation.MobileView.Link /> should render base component correctly 1`] = `
<div>
  <a
    class="base left dark base dark itemElement"
    href="#link"
    tabindex="-1"
    target="_self"
    title="Link"
  >
    Link
  </a>
</div>
`;
